cmd/flags_core.go

```diff
@@ -12,70 +12,78 @@ import (
 )
 
 var (
-	coreRemoteFlag = "core.remote"
-	coreGRPCFlag   = "core.grpc"
+	coreFlag     = "core.ip"
+	coreRPCFlag  = "core.rpc.port"
+	coreGRPCFlag = "core.grpc.port"
 )
 
 // CoreFlags gives a set of hardcoded Core flags.
 func CoreFlags() *flag.FlagSet {
 	flags := &flag.FlagSet{}
 
 	flags.String(
-		coreRemoteFlag,
+		coreFlag,
 		"",
-		"Indicates node to connect to the given remote core node. "+
-			"Example: <protocol>://<ip>:<port>, tcp://127.0.0.1:26657",
+		"Indicates node to connect to the given core node. "+
+			"Example: <ip>, 127.0.0.1. Assumes RPC port 26657 and gRPC port 9009 as default unless otherwise specified.",
+	)
+	flags.String(
+		coreRPCFlag,
+		"26657",
+		"Set a custom RPC port for the core node connection. The --core.ip flag must also be provided.",
 	)
 	flags.String(
 		coreGRPCFlag,
-		"",
-		"Indicates node to connect to the given celestia-app gRPC endpoint for state-related queries"+
-			"Example: <ip>:<port>, 127.0.0.1:9090",
+		"9090",
+		"Set a custom gRPC port for the core node connection. The --core.ip flag must also be provided.",
 	)
 
 	return flags
 }
 
 // ParseCoreFlags parses Core flags from the given cmd and applies values to Env.
 func ParseCoreFlags(cmd *cobra.Command, env *Env) error {
-	coreRemote := cmd.Flag(coreRemoteFlag).Value.String()
-	if coreRemote != "" {
-		proto, addr, err := validateAddress(coreRemote)
+	coreIP := cmd.Flag(coreFlag).Value.String()
+	if coreIP != "" {
+		// sanity check given core ip addr and strip leading protocol
+		ip, err := sanityCheckIP(coreIP)
 		if err != nil {
-			return fmt.Errorf("cmd: while parsing '%s': %w", coreRemoteFlag, err)
+			return err
 		}
 
-		env.AddOptions(node.WithRemoteCore(proto, addr))
-	}
+		rpc := cmd.Flag(coreRPCFlag).Value.String()
+		// sanity check rpc endpoint
+		_, err = strconv.Atoi(rpc)
+		if err != nil {
+			return err
+		}
+		env.AddOptions(node.WithRemoteCoreIP(ip), node.WithRemoteCoreRPCPort(rpc))
 
-	grpcEndpoint := cmd.Flag(coreGRPCFlag).Value.String()
-	if grpcEndpoint != "" {
-		// parse ip:port from the endpoint
-		grpcURL, err := url.Parse(grpcEndpoint)
+		grpc := cmd.Flag(coreGRPCFlag).Value.String()
+		// sanity check gRPC endpoint
+		_, err = strconv.Atoi(grpc)
 		if err != nil {
 			return err
 		}
-		env.AddOptions(node.WithGRPCEndpoint(grpcURL.Host))
+		env.AddOptions(node.WithGRPCPort(grpc))
+		return nil
+	}
+	if cmd.Flag(coreGRPCFlag).Changed || cmd.Flag(coreRPCFlag).Changed {
+		return fmt.Errorf("cannot specify RPC/gRPC ports without specifying an IP address for --core.ip")
```

Comment:
Nice. The only problem here is that any new added core flag parsing logic would need to be nested in the if above or coreIP plag to be always In the end, which is fine

Reply:
https://github.com/celestiaorg/celestia-node/pull/910/commits/8ccb595089e1d071c1c0bd9a40ddce1c2d7bd769